### PR TITLE
Add disable_outbound_snat configuration to azure_rm_loadbalancer module

### LINKED ISSUE #743: azure_rm_loadbalancer does not expose disable_outbound_snat option (LoadBalancingRuleMustDisableSNATSinceSameFrontendIPConfigurationIsReferencedByOutboundRule)
##### SUMMARY
I would like to be able to configure `disable_outbound_snat` (maps to `disableOutboundSNAT`) when setting up Load Balancer rules via `azure_rm_loadbalancer` module. 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
`azure_rm_loadbalancer`

##### ADDITIONAL INFORMATION

Currently I get the following error when trying to add a new loadbalancing rule:

```
fatal: [localhost]: FAILED! => {
  "changed": false, 
  "msg": "Error creating or updating load balancer dummy-load-balancer - Azure Error: LoadBalancingRuleMustDisableSNATSinceSameFrontendIPConfigurationIsReferencedByOutboundRule\nMessage: Load Balancing Rules /subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/dev-backend/providers/Microsoft.Network/loadBalancers/dummy-load-balancer/loadBalancingRules/tradefeedr must have disableOutboundSNAT set to true when same FrontendIPConfiguration /subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/dev-backend/providers/Microsoft.Network/loadBalancers/dummy-load-balancer/frontendIPConfigurations/dummy-load-balancer-ip is referenced by Outbound Rules /subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/dev-backend/providers/Microsoft.Network/loadBalancers/dummy-load-balancer/outboundRules/SNAT"
}
```

Telling me that I **must have disableOutboundSNAT set to true**...

I have already made a fix locally, so will submit a PR shortly.

### PR DESCRIPTION
**Pull request by tfmark**
##### SUMMARY
Fixes #743

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`azure_rm_loadbalancer`

##### ADDITIONAL INFORMATION

```yaml
  - name: Add Load Balancer rules
    azure.azcollection.azure_rm_loadbalancer:
      profile: prod
      resource_group: dummy-resource-group
      name: dummy-load-balancer
      load_balancing_rules :
        - name: dummy-rule-1
          backend_port: 9999
          protocol: Tcp
          frontend_port: 9999
          frontend_ip_configuration: dummy-load-balancer-ip
          backend_address_pool : dummy-load-balancer-pool
          probe: dummy-health-probe
          disable_outbound_snat: true # << new feature
```

### CODE REVIEW

**Comment by Fred-sun** [comment]
@tfmark Can you add a test case for this newly added parameter? Thank you very much!

File: plugins/modules/azure_rm_loadbalancer.py [hunk 1]
```diff
@@ -207,6 +207,12 @@
             enable_floating_ip:
                 description:
                     - Configures a virtual machine's endpoint for the floating IP capability required to configure a SQL AlwaysOn Availability Group.
+            disable_outbound_snat:
+                description:
+                    - Configure outbound source network address translation (SNAT)
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_loadbalancer.py]:
```suggestion
                    - Configure outbound source network address translation (SNAT).
```

File: plugins/modules/azure_rm_loadbalancer.py [hunk 2]
```diff
@@ -207,6 +207,12 @@
...
+                    - The default value is False
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_loadbalancer.py]:
```suggestion
                    - The default value is I(disable_outbound_snat=False).
```

File: plugins/modules/azure_rm_loadbalancer.py [hunk 3]
```diff
@@ -207,6 +207,12 @@
...
+                    - True is equivalent to "Use outbound rules to provide backend pool members access to the internet" in portal
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_loadbalancer.py]:
```suggestion
                    - True is equivalent to "Use outbound rules to provide backend pool members access to the internet" in portal.
```

File: plugins/modules/azure_rm_loadbalancer.py [hunk 4]
```diff
@@ -207,6 +207,12 @@
...
+                    - False is equivalent to "Use implicit outbound rule" in portal
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_loadbalancer.py]:
```suggestion
                    - False is equivalent to "Use implicit outbound rule" in portal.
```

**Comment by Fred-sun** [comment]
@tfmark Are you still contributing to this PR? Please help rebase this PR and fix the problems in the comments above! Thank you very much!

**Comment by tfmark** [comment]
Hi @Fred-sun, I will look into this, but this is something work-related rather than something I do for fun. Last time I made a PR I found that it was difficult to test, but maybe things have changed.

I'll get around to it at some point.

**Comment by Fred-sun** [comment]
@tfmark It is not recommended to set a default value for this parameter, because if the resource already exists and we create it again, if we do not set this parameter, it will be updated to the default value. But we can document that the default is true or false. Thank you very much!

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**